src/core/settings/qgssettingsentry.cpp

```diff
@@ -398,11 +504,20 @@ QgsSettingsEntryDouble::QgsSettingsEntryDouble( const QString &key, const QStrin
 }
 
 bool QgsSettingsEntryDouble::setValue( double value, const QString &dynamicKeyPart ) const
+{
+  QStringList dynamicKeyPartList;
+  if ( !dynamicKeyPart.isEmpty() )
+    dynamicKeyPartList.append( dynamicKeyPart );
```

Comment:
isn't an empty string already a value?
this will not cause an issue?

Reply:
This would be an issue if someone wants to just remove the dynamic part by replacing it with an empty string. I will change the check with isNull()